test/e2e/network/network_policy.go

```diff
@@ -1744,3 +1747,69 @@ func cleanupNetworkPolicy(f *framework.Framework, policy *networkingv1.NetworkPo
 		framework.Failf("unable to cleanup policy %v: %v", policy.Name, err)
 	}
 }
+
+var _ = SIGDescribe("SCTP NetworkPolicy [Feature:SCTP] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("sctpnp")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		// SCTP in K8s is not supported on Windows
+		e2eskipper.SkipIfNodeOSDistroIs("windows")
+
+		if proxyMode, err := proxyMode(f); err == nil {
+			if proxyMode == "userspace" {
+				e2eskipper.Skipf("The test doesn't work with kube-proxy in userspace mode")
+			}
+		} else {
+			framework.Logf("Couldn't detect KubeProxy mode - test failure may be expected: %v", err)
+		}
+
+		cs = f.ClientSet
+	})
+
+	ginkgo.It("should allow acces only for SCTP [Feature:NetworkPolicy]", func() {
+		ginkgo.By("getting the state of the sctp module on nodes")
+		nodes, err := e2enode.GetReadySchedulableNodes(cs)
+		framework.ExpectNoError(err)
+		sctpLoadedAtStart := VerifySCTPModuleLoadedOnNodes(f, nodes)
+
+		ginkgo.By("creating a policy that allows only SCTP traffic.")
+		protocolSCTP := v1.ProtocolSCTP
+		policy := &networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "allow-only-sctp",
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				// Apply this policy to client-a
+				PodSelector: metav1.LabelSelector{},
+				// Allow traffic to only one port: 5060.
```

Comment:
Likewise.

Also, if we only have one SCTP policy, I wouldn't test Ingress and Egress together, and I wouldn't test the weird "all ports" thing. I'd create a policy that allows ingress to a single SCTP port.

Reply:
Fixed.